tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -148,12 +149,33 @@ HloInstruction *MaybeConstantFoldBias(HloInstruction *bias) {
 // and provided C has no other users).
 // We then guide the buffer assignment to alias the buffer of the custom call
 // and C.
+//
+// For scaled FP8 GEMMs on Hopper systems, the following steps mentioned in RFC
+// #22 (https://github.com/openxla/xla/discussions/22) are elided and rewritten
+// into a Custom Call:
+//
+// 1. Cast each input from FP8 to a wider type such as FP16 or FP32.
+// 2. Unscale each input by multiplying each input by the corresponding input
+// scale.
+// 3. Evaluate the matrix multiplication on the scaled inputs.
+// 4. Compute the maximum of the absolute values in the result of the GEMM
+// (DAmax).
+// 5. Scale the output by dividing the output by the output scale.
+// 6. Cast the output back to FP8. Since saturation should be done on overflow,
+// this is represented by a Clamp instruction followed by a Convert instruction.
+
+// Steps 1 through 3 can be elided independently of the remainder. Steps 5 and 6
+// can be elided only if steps 1 through 3 were successfully transformed. Step 4
+// requires steps 5 and 6, i.e. the computation of DAmax can be elided only when
+// the output of the GEMM is requested in FP8 format.
```

Comment:
Do steps 5 and 6 alone make sense? You could have just 3, GEMM on FP8 inputs with a_scale & b_scale = 1 returning a result in a wide type, and then potentially add 5 and 6. Enabling this case seems to be fairly straightforward, I’m just not sure it would be meaningful.

Reply:
Yeah I meant steps 3, 5, and 6 alone make sense (you always need 3). Anyway I'm happy with the current wording now.